manifests/kustomize/third-party/metacontroller/base/stateful-set.yaml

```diff
@@ -18,26 +18,28 @@ spec:
         sidecar.istio.io/inject: "false"
     spec:
       containers:
-      - command:
-        - /usr/bin/metacontroller
-        - --logtostderr
-        - -v=4
-        - --discovery-interval=20s
-        image: metacontroller/metacontroller:v0.3.0
-        imagePullPolicy: Always
-        name: metacontroller
-        ports:
-        - containerPort: 2345
-        resources:
-          limits:
-            cpu: "4"
-            memory: 4Gi
-          requests:
-            cpu: 500m
-            memory: 1Gi
-        securityContext:
-          allowPrivilegeEscalation: true
-          privileged: true
+        - resources:
+            limits:
+              cpu: '1'
+              memory: 1Gi
+            requests:
+              cpu: 100m
+              memory: 256Mi
+          command:
+            - /usr/bin/metacontroller
+            - --zap-log-level=4
+            - '--discovery-interval=3600s' # less insane than 10 seconds
+          securityContext:
+            capabilities:
+              drop:
+              - ALL
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            runAsUser: 1000
```

Comment:
Any reason for the security context setup here?

Reply:
yes, we obviously do want to run metacontroller with the smallest amount of permissions so an unpriviledged non-root. I can remove runAsUser: 1000 if you think that it is too much.

Reply:
As you can see in the full source code it had a securitycontext before and was a privileged root container before.

Reply:
Got it, thank you for the explanation, we can keep runAsUser and we can come back if encountering any issue.